### DISCUSSION TITLE
Unsupported TuYa power strip

### DISCUSSION THREAD

**Discussion by mircicd** [Koenkk/zigbee2mqtt#22356] [Devices]
Hey there!

I've been a very happy user of Z2M for a while now, thanks a lot for the great work! I'm mostly using zigbee devices that are supported by Z2M out of the box with some small tweaks using external converters here and there.

Recently I purchased a power strip whose full functionality isn't supported and I'm struggling to get it working myself.
This is the device: https://www.aliexpress.com/item/1005006744575899.html
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1148613/a5012246-cba6-481e-b0c7-12d8682c6855)

It does get recognized and automatically added to Z2M as the following:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1148613/cb1cc0d1-2b7d-4565-821a-454e5c6c5b27)

The manufacturer ID is: `_TZ3000_gdyjfvgm`
It looks like it is being recognized as a single power plug. A single switch is being exposed. Toggling it toggles the first socket of the strip. I'm unable to control the remaining 3 (and one USB strip) sockets, as they are not being recognized.

I'm not sure if this strip is just a series of single `TS011F` switches built into one strip, or if it is a whole new device. There is a physical button on the strip that toggles all switches in a weird consecutive order (one after another) with a short delay. This is the log output during that operation:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1148613/63a7cf00-3495-47fb-b7e5-4e5d2b802bc8)

I'm quite the newbie with endpoints but I'm guessing the way going forward here is to properly `expose` them in an external converter. Any support is greatly appreciated!

Br, Dejan

**Comment by mircicd** [comment]
I managed to set it up successfully, using the following external converter .js:
```
const definition = {
    // Since a lot of TuYa devices use the same modelID, but use different datapoints
    // it's necessary to provide a fingerprint instead of a zigbeeModel
    fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0601' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS011F',
            // The manufacturer name from: Device with modelID 'TS0601' is not supported.
            manufacturerName: '_TZ3000_gdyjfvgm',
        },
    ],
    model: 'TS011F_5',
    vendor: 'TuYa',
    description: '5 gang switch',
    extend: [tuya.modernExtend.tuyaOnOff({powerOutageMemory: true, childLock: true, endpoints: ['l1', 'l2', 'l3', 'l4', 'l5']})],
    endpoint: (device) => {
        return {l1: 1, l2: 2, l3: 3, l4: 4, l5: 5};
    },
    meta: {multiEndpoint: true},
    configure: async (device, coordinatorEndpoint) => {
        await tuya.configureMagicPacket(device, coordinatorEndpoint);
        for (const ID of [1, 2, 3, 4, 5]) {
            await reporting.bind(device.getEndpoint(ID), coordinatorEndpoint, ['genOnOff']);
        }
    }
};
```

**Comment by Koenkk** [comment]
Added, thanks!

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.
